packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -246,12 +258,23 @@ class BuildInfo {
     };
   }
 
+  Map<String, dynamic>? defineConfigJsonRawMap() {
+    if (defineConfigJsonMap != null) {
+      return defineConfigJsonMap;
+    }
+    return <String, dynamic>{};
+  }
+
   /// Convert to a structured string encoded structure appropriate for usage as
   /// environment variables or to embed in other scripts.
   ///
   /// Fields that are `null` are excluded from this configuration.
   Map<String, String> toEnvironmentConfig() {
-    return <String, String>{
+    final Map<String, String> map = <String, String>{};
+    defineConfigJsonRawMap()!.forEach((String key, dynamic value) {
+      map[key]='$value';
+    });
+    map.addAll(<String, String>{
       if (dartDefines.isNotEmpty)
         'DART_DEFINES': encodeDartDefines(dartDefines),
```

Comment:
If the user sends dart defines variables on their json file we would override them in here which is fine but we need to let them know when that happens so no funny issues arrises after this. please add a warning whenever this happens

Reply:
OK,I add a warning prompt message